### PR TITLE
chore: git mod - adding git ops modal

### LINKED ISSUE #37803: [Task] Git mod - Components - Commit input & button

### LINKED ISSUE #37804: [Task] Git mod - Components - Status display

### LINKED ISSUE #37807: [Task] Git mod - Components - Discard flow

### LINKED ISSUE #37826: [Task] Git mod - Components - Git Merge

### PR DESCRIPTION
**Pull request by brayn003**
## Description
- Adds git modal
- Adds pull saga
- Adds ability to commit
- Adds ability to merge
- Adds ability to see status
- Adds ability to discard
- Improves on saga error handling

Fixes #37826 
Fixes #37803 
Fixes #37804 
Fixes #37807

## Automation

/ok-to-test tags="@tag.Git"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/12319147488>
> Commit: ec461fde26da90af2f365fb36f49e3758c14cb96
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=12319147488&attempt=2" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Git`
> Spec:
> <hr>Fri, 13 Dec 2024 17:27:36 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced new components for managing Git operations, including `ConflictErrorView`, `OpsModal`, `TabDeploy`, `TabMerge`, and `StatusChanges`.
	- Added loading and error handling components such as `StatusLoader` and `DiscardFailedError`.
	- Enhanced the `StatusTree` component for displaying hierarchical status changes.
- **Bug Fixes**
	- Improved error handling across various sagas and components.
- **Refactor**
	- Renamed several modal toggle functions for clarity and consistency.
	- Updated prop types for improved type safety in components.
- **Chores**
	- Removed unused component `GitTest`.
	- Updated imports and exports for better organization.

### CODE REVIEW

File: app/client/src/git/components/GitOpsModal/TabDeploy/index.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      isFetchStatusLoading={fetchStatusLoading}
+      isPullFailing={isPullFailing}
+      isPullLoading={pullLoading}
+      lastDeployedAt={lastDeployedAt}
+      pull={pull}
+      remoteUrl={remoteUrl}
+      statusBehindCount={statusBehindCount}
+      statusIsClean={statusIsClean}
+    />
+  );
```
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/src/git/components/GitOpsModal/TabDeploy/index.tsx]: its already memoized

File: app/client/src/git/components/GitOpsModal/TabDeploy/DeployPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    ? `${createMessage(LATEST_DP_SUBTITLE)} ${howMuchTimeBeforeText(
+        lastDeployedAt,
+        {
+          lessThanAMinute: true,
+        },
+      )} ago`
+    : "";
+
+  return lastDeployedAt ? (
+    <Container className="t--git-deploy-preview">
```
Comment (ashit-rath) [inline app/client/src/git/components/GitOpsModal/TabDeploy/DeployPreview.tsx:64]: The `Flex` component can be used here but I'm assuming you have noted all the refactors and would be taking care of it in the next PR(s)?

File: app/client/src/git/components/GitOpsModal/TabDeploy/SubmitWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import noop from "lodash/noop";
+import React, { useCallback } from "react";
+import { isMacOrIOS } from "utils/helpers";
+
+interface SubmitWrapperProps {
+  children: React.ReactNode;
+  onSubmit: () => void;
+}
+
+export default function SubmitWrapper({
```
Comment (ashit-rath) [inline app/client/src/git/components/GitOpsModal/TabDeploy/SubmitWrapper.tsx:10]: Can this be converted to a utility function or a hook?

File: app/client/src/git/components/GitQuickActions/DumbGitQuickActions.tsx [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    AnalyticsUtil.logEvent("GS_CONNECT_GIT_CLICK", {
+      source: "BOTTOM_BAR_GIT_CONNECT_BUTTON",
+    });
+
+    toggleConnectModal(true);
+  }, [toggleConnectModal]);
+
+  return isGitConnected ? (
+    <Container>
+      {/* <BranchButton /> */}
```
Comment (ashit-rath) [inline app/client/src/git/components/GitQuickActions/DumbGitQuickActions.tsx:129]: Have we moved this button to uncomment this?

File: app/client/src/git/components/GitConflictErrorModal/DumbGitConflictErrorModal.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        hasBackdrop
+        isOpen={isConflictErrorModalOpen}
+        onClose={handleClose}
+        transitionDuration={25}
+        usePortal={false}
+      >
+        <div className={Classes.OVERLAY_CONTENT}>
+          <div className="git-error-popup">
+            <div
+              style={{
```
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/src/git/components/GitConflictErrorModal/DumbGitConflictErrorModal.tsx:66]: Just copy pasted this from the old files, I have taken a not, will take a look at it in the subsequent PRs

File: app/client/src/git/components/GitOpsModal/TabDeploy/DeployPreview.tsx (hunk 2 above)
Comment (brayn003) [reply app/client/src/git/components/GitOpsModal/TabDeploy/DeployPreview.tsx:64]: I have taken a not, will take a look at it in the subsequent PRs

File: app/client/src/git/components/GitOpsModal/TabDeploy/SubmitWrapper.tsx (hunk 3 above)
Comment (brayn003) [reply app/client/src/git/components/GitOpsModal/TabDeploy/SubmitWrapper.tsx:10]: Just a copy paste from the prev files, will take a look at it later

**Pull request merged**